### PR TITLE
[PAL] replace `PAL_GET_TYPE ` macros with inline code.

### PR DESCRIPTION
**Pull request by billionairiam**
## Description of the changes 
Remove `PAL_GET_TYPE ` macros and replace with inline code

---
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/gramine/755)

### CODE REVIEW

**Review by boryspoplawski — APPROVED** [review]
> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/755)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @dimakuv)

**Review by boryspoplawski — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/755)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @dimakuv)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/755#-N6wYBn2-5xgrc-b3qJq:-N6wYBn3C9nbqEaAEdKQ:b8t21uo) (no related file):*
When you submit PRs, please base them on the current, up-to-date master. Base of this is 27 commits behind.

___

**Comment by billionairiam** [comment]
> > _[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/755)_ status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @dimakuv)
> 
> _[a discussion](https://reviewable.io/reviews/gramineproject/gramine/755#-N6wYBn2-5xgrc-b3qJq:-N6wYBn3C9nbqEaAEdKQ:b8t21uo) (no related file):_ When you submit PRs, please base them on the current, up-to-date master. Base of this is 27 commits behind.
Can I follow these steps to commit a PR?
1. Sync the repo again
    $ git fetch origin
    $ git rebase origin/master
2. create the branch and make changes
    $ git branch <api_change>
    $ git checkout <api_change>
    // make changes in your branch
    $ git add .
    $ git commit -s -m "...."
3. Sync the repo again
    $ git fetch origin
    $ git rebase origin/master
4. push to remote repo
    $ git push -u origin <api_change>
5. goto web page and submit the merge request

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/755)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @dimakuv)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/755#-N6wYBn2-5xgrc-b3qJq:-N7-diuK4z2HUzOr6926:bd9i75t) (no related file):*
<details><summary><i>Previously, billionairiam (Liang Ma) wrote…</i></summary><blockquote>

Can I follow these steps to commit a PR?
1. Sync the repo again
    $ git fetch origin
    $ git rebase origin/master
2. create the branch and make changes
    $ git branch <api_change>
    $ git checkout <api_change>
    // make changes in your branch
    $ git add .
    $ git commit -m -s "...."
3. Sync the repo again
    $ git fetch origin
    $ git rebase origin/master
4. push to remote repo
    $ git push -u origin <api_change>
5. goto web page and submit the merge request

</blockquote></details>

You should remove step 3 from this sequence. The rest is correct: you first fetch the latest origin/master, then you create your new branch, then you add/modify the files and commit them, then you push to the remote repo.

In other words, the step `git rebase origin/master` should be done **only once** -- as the very first step when creating a new PR.

___

**Review by billionairiam — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/755)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @dimakuv)

___
*[`-- commits` line 4 at r1](https://reviewable.io/reviews/gramineproject/gramine/755#-N6vdgl50Mbol_xAyBzT):*
<details><summary><i>Previously, billionairiam (Liang Ma) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Thanks!!

**Review by boryspoplawski — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/755)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @dimakuv)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/755#-N6wYBn2-5xgrc-b3qJq:-N70pkY9-ZDeNbeq3hlg:b-t28yfo) (no related file):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

You should remove step 3 from this sequence. The rest is correct: you first fetch the latest origin/master, then you create your new branch, then you add/modify the files and commit them, then you push to the remote repo.

In other words, the step `git rebase origin/master` should be done **only once** -- as the very first step when creating a new PR.
</blockquote></details>

@dimakuv Why remove it? It was correct, you should make sure that your base is up-to-date when submitting a PR. Do not rebase only **after** you submitted the PR.

___

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/755)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @dimakuv)

*[a discussion](https://reviewable.io/reviews/gramineproject/gramine/755#-N6wYBn2-5xgrc-b3qJq:-N7F-mcGA8ctiaMiYnZ_:b-lvf4kz) (no related file):*
<details><summary><i>Previously, boryspoplawski (Borys Popławski) wrote…</i></summary><blockquote>

@dimakuv Why remove it? It was correct, you should make sure that your base is up-to-date when submitting a PR. Do not rebase only **after** you submitted the PR.
</blockquote></details>

@boryspoplawski Ah, I see. In this case yes. Right before submitting a PR, it is good to make a `git rebase origin/master`. But after submitting a PR, one should *not* do `git rebase origin/master` anymore.

___

**Review by dimakuv — APPROVED** [review]
> Reviewed 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/755)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: Intel)

**Comment by dimakuv** [comment]
Jenkins, test this please

**Review by mkow — APPROVED** [review]
> Reviewed 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/755)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by boryspoplawski — APPROVED** [review]
> Reviewed 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/755)* status: :shipit: complete! all files reviewed, all discussions resolved
